leptonai/cli/pod.py

```diff
@@ -174,20 +174,31 @@ def list_command(pattern):
         return 0
     ssh_ports = []
     tcp_ports = []
+    tcp_ports_jupyterlab = []
     for pod in pods:
         ports = pod.spec.container.ports
         port_pairs = [(p.container_port, p.host_port) for p in ports]
         check(
-            len(port_pairs) == 2,
-            f"Pod {pod.metadata.name} does not have exactly two ports. This is not"
-            " supported.",
+            len(port_pairs) in [2, 3],
+            f"Pod {pod.metadata.name} does not have exactly two or three ports. This is"
+            " not supported.",
         )
-        if port_pairs[0][0] == 2222:
-            ssh_ports.append(port_pairs[0])
-            tcp_ports.append(port_pairs[1])
-        else:
-            ssh_ports.append(port_pairs[1])
-            tcp_ports.append(port_pairs[0])
+
+        if len(port_pairs) == 2:
+            tcp_ports_jupyterlab.append(None)
+        for port_pair in port_pairs:
+            if port_pair[0] == SSH_PORT:
+                ssh_ports.append(port_pair)
+            elif port_pair[0] == TCP_PORT:
+                tcp_ports.append(port_pair)
+            elif port_pair[0] == TCP_JUPYTER_PORT:
+                tcp_ports_jupyterlab.append(port_pair)
```

Comment:
still not fixed. For example, if the port array had 3 elements, including an ssh port, a TCP port, and a random other port (no jupyter port), the output would still be messed up

Reply:
Hi Cong,

Just to clarify, the only valid port pairs are:

Length = 2: [SSH, TCP]
Length = 3: [SSH, TCP, Jupyter]

For all other cases, the program should sys.exit and display an error message. Is that correct?

Reply:
if the data form backend is [SSH, TCP, SOMETHING_RANDOM], without Jupyter, cli should either display and error message or display nothing, or only display SSH, TCP. But cli should not mess up all the data (it seems this PR would do so)

Reply:
or even worse, the resp is [SSH, SOMETHING_RANDOM, SOMETHING_RANDOM_2]

Reply:
sys.exit is not ideal either because a single bad pod should not prevent customers from listing other pods